engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -105,9 +105,15 @@ object MaterializeWorkflowDescriptorActor {
       }
     }
 
-    val enabled = conf.as[Option[Boolean]]("call-caching.enabled").getOrElse(false)
-    val invalidateBadCacheResults = conf.as[Option[Boolean]]("call-caching.invalidate-bad-cache-results").getOrElse(true)
-    if (enabled) {
+    val callCachingConfig = conf.getConfig("call-caching")
+
+    def errorOrCallCachingBoolean(path: String): ErrorOr[Boolean] = {
+      import common.validation.Validation._
+      validate(callCachingConfig.getBoolean(path))
```

Comment:
I don't understand why you switched from `Option` to `ErrorOr` to indicate "found or not found" here?

Reply:
See the new tests. All the configs are using `ErrorOr`s now. When misconfigured with something like `my-config-enabled = tru`, a naked `getBoolean` will quick-crash instead of continuing to validate other config values.